### PR TITLE
Add option to disable rate limiting via AUTH_RATELIMIT=False

### PR DESCRIPTION
**Pull request by lub**
## What type of PR?

Feature, enhancement

### Related issue(s)
- maybe this conflicts with the changes in #1360, although I think #1360 would need to be rebased anyway

## Prerequistes
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/guide.html#changelog) entry file.

### CODE REVIEW

File: core/admin/mailu/internal/views/auth.py [hunk 1]
```diff
@@ -12,24 +12,26 @@
...
-    if not limiter.test(client_ip):
-        response = flask.Response()
-        response.headers['Auth-Status'] = 'Authentication rate limit from one source exceeded'
-        response.headers['Auth-Error-Code'] = '451 4.3.2'
-        if int(flask.request.headers['Auth-Login-Attempt']) < 10:
-            response.headers['Auth-Wait'] = '3'
-        return response
     headers = nginx.handle_authentication(flask.request.headers)
     response = flask.Response()
+    ratelimit = app.config["AUTH_RATELIMIT"]
```
Comment (nextgens) [inline core/admin/mailu/internal/views/auth.py:17]: Rate-limiting after the actual password check is a big no-no, I've proposed an alternative implementation in #1745

**Review by nextgens — CHANGES_REQUESTED** [review]
The expensive call is nginx.handle_authentication(flask.request.headers) ... this shouldn't run post rate limiting

File: docs/configuration.rst [hunk 2]
```diff
@@ -40,6 +40,7 @@ address.
 The ``AUTH_RATELIMIT`` holds a security setting for fighting attackers that
 try to guess user passwords. The value is the limit of failed authentication attempts
 that a single IP address can perform against IMAP, POP and SMTP authentication endpoints.
+Setting this this to ``False`` disables all rate limiting.
```
Comment (Diman0) [inline docs/configuration.rst:43]: You want to remove the duplicate "this" in this sentence.

**Review by Diman0 — CHANGES_REQUESTED** [review]
Correct the typo in the doc. I think we can approve the PR after this. 
At a later moment we can look at the changes suggested by Nextgens.

**Comment by nextgens** [comment]
IMHO this has very little added value over putting a very high limit. If you want it disabled, set  it to 10000/second or something.

**Comment by nextgens** [comment]
This has been deprecated by #1910; more work will happen on rate limiting in next releases

**Pull request closed without merging**